### PR TITLE
fix(combobox): include groups in filtering

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #7702 

## Summary

Includes groups while filtering in `calcite-combobox`.

### CODE REVIEW

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts [hunk 1]
```diff
@@ -534,6 +534,62 @@ describe("calcite-combobox", () => {
...
+          </calcite-combobox-item-group>
+        </calcite-combobox>`,
+      );
+
+      const combobox = await page.find("calcite-combobox");
+      await combobox.click();
+      await page.waitForChanges();
+      await combobox.type("group");
+      await page.waitForChanges();
+      await new Promise((res) => setTimeout(() => res(true), DEBOUNCE.filter));
```
Comment (jcfranco) [inline packages/calcite-components/src/components/combobox/combobox.e2e.ts:559]: For consistency, can you use `await page.waitForTimeout`? Applies to other timeouts.

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts [hunk 2]
```diff
@@ -534,6 +534,62 @@ describe("calcite-combobox", () => {
...
+
```
Comment (jcfranco) [inline packages/calcite-components/src/components/combobox/combobox.e2e.ts:560]: Would waiting for `calciteComboboxFilterChange` help avoid the debounce timeout? We can revisit this separately.

**Review by jcfranco — APPROVED** [review]
👨‍🚀🚀🪐

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts (hunk 2 above)
Comment (anveshmekala) [reply packages/calcite-components/src/components/combobox/combobox.e2e.ts:560]: Agree a refactor PR would be beneficial considering it is repeated multiple times.

**Pull request merged**